blog_server.js

```diff
@@ -0,0 +1,168 @@
+"use strict";
+
+const _            = require('lodash');
+const express      = require('express');
+const bodyParser   = require('body-parser');
+const config  = require('./knexfile.js');
+
+// Initialize Express.
+const app = express();
+app.use(bodyParser.json());
+
+// Configure & Initialize Bookshelf & Knex.
+console.log('Running in environment: ' + process.env.NODE_ENV);
+const knex = require('knex')(config[process.env.NODE_ENV]);
+const bookshelf = require('bookshelf')(knex);
+
+// ***** Models ***** //
+
+const User = bookshelf.Model.extend({
+  tableName: 'users',
+  hasTimestamps: true,
+  posts: function() {
+    return this.hasMany(Posts, 'author');
+  },
+  comments: function() {
+    return this.hasMany(Comments);
+  },
+});
+
+const Posts = bookshelf.Model.extend({
+  tableName: 'posts',
+  hasTimestamps: true,
+  author: function() {
+    return this.belongsTo(User, 'author');
+  },
+  comments: function() {
+    return this.hasMany(Comments);
+  },
+});
+
+const Comments = bookshelf.Model.extend({
+  tableName: 'comments',
+  hasTimestamps: true,
+  user: function() {
+    return this.belongsTo(User);
+  },
+  post: function() {
+    return this.belongsTo(Posts);
+  },
+});
+
+exports.User = User;
+exports.Posts = Posts;
+exports.Comments = Comments;
+
+// ***** Server ***** //
+
+app.get('/user/:id', (req,res) => {
+  User
+    .forge({id: req.params.id})
+    .fetch()
+    .then((usr) => {
+      if (_.isEmpty(usr))
+        return res.sendStatus(404);
+      res.send(usr);
+    })
+    .catch((error) => {
+      console.error(error);
+      return res.sendStatus(500);
+    });
+});
+
+app.post('/user', (req, res) => {
+  if (_.isEmpty(req.body))
+    return res.sendStatus(400);
```

Comment:
Just a style nit: I think we encourage students to use curly braces around the bodies of `if` statements; but they should understand what this means.